### ISSUE TITLE
When I do training, why are the results of the validation set obtained by using the teacher model?

### ISSUE THREAD

**Issue by wangclnlp** [LeslieOverfitting/selective_distillation#1]
021-07-18 18:20:35 | INFO | valid | epoch 006 | valid on 'valid' subset | loss 3.932 | nll_loss 2.125 | distil_rate 0 | ppl 4.36 | bleu 42.08 | wps 10164.3 | wpb 10722.3 | bsz 332.8 | num_updates 15150 | best_bleu 42.08

**Comment by wangclnlp** [comment]
It is solved, thanks.

**Comment by FadedCosine** [comment]
> 

how can you solve this issue?

**Comment by Chenglong-coder** [comment]
> > 
> 
> how can you solve this issue?

Firstly, after I use the loss metric, this error will been removed.

**Comment by Chenglong-coder** [comment]
> > 
> 
> how can you solve this issue?

Firstly, after I use the loss metric, this error will been removed.

**Comment by wangclnlp** [comment]
> > 
> 
> how can you solve this issue?

Firstly, after I use the loss metric, this error will been removed.    ( --best-checkpoint-metric loss)    Besides that, I rewrote some 
rows of fairseq_cli/train.py  to remove this error radically.   I will send the file to you by using email.  Hope it is helpful to you!

**Comment by Chenglong-coder** [comment]
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "LeslieOverfitting/selective_distillation"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年12月22日(星期三) 晚上8:48
***@***.***&gt;;
抄送:&nbsp;"wangchenglong ***@***.***&gt;;"State ***@***.***&gt;;
主题:&nbsp;Re: [LeslieOverfitting/selective_distillation] When I do training, why are the results of the validation set obtained by using the teacher model? (#1)





   
how can you solve this issue?
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by FadedCosine** [comment]
> 

Thank you, I have moved line 117 of fairseq_cli/train.py to line 67 (before " model = task.build_model(args) "). And then it works well (in a ugly way).

**Comment by TobiasLee** [comment]
@Chenglong-coder  Met the same issue, could you share the code snippet? or maybe made a PR to the repo? Thanks!
